main.py

```diff
@@ -759,20 +763,50 @@ def handle_link_up(self, event):
 
     # Possibly replace this with interruptions?
     @listen_to(
-        '.*.switch.interface.(link_up|link_down|created|deleted)',
-        pool='dynamic_single'
+        '.*.switch.interface.(link_up|link_down|created|deleted)'
     )
     def on_interface_link_change(self, event: KytosEvent):
         """
-        Handler for interface link_up and link_down events
+        Handler for interface link_up and link_down events.
+
+        To avoid multiple database updated (link flap):
+        First check the time when interface was last updated.
+        Second check the time when the last interface event was received.
+        Last event received always updates database.
         """
-        with self._lock:
-            _, _, event_type = event.name.rpartition('.')
-            iface = event.content.get("interface")
-            if event_type in ('link_up', 'created'):
-                self.handle_interface_link_up(iface)
-            elif event_type in ('link_down', 'deleted'):
-                self.handle_interface_link_down(iface)
+        iface = event.content.get("interface")
+        self.intf_events[iface.id].update({"received": now()})
+        with self.lock_interfaces[iface.id]:
+            last_updated = self.intf_events[iface.id].get("updated")
+            _now = now()
+            if not last_updated:
+                self.intf_events[iface.id].update({"updated": _now})
+                self.handle_on_interface_link_change(event)
+            elif int((_now - last_updated).total_seconds() * 1000) >= 100:
+                last_updated = last_updated.replace(tzinfo=timezone.utc)
+                self.intf_events[iface.id].update({"updated": _now})
+                self.handle_on_interface_link_change(event)
+
+        time.sleep(0.1)
```

Comment:
0.1 secs is very reasonable let's make it configurable on `settings.py`, and based on how you've explored let's also briefly document how it performed watching out the APM and seeing the hits in the database. Let's compare once you're done with the implementation if with 0.1 secs is amortizing as we expected compared with a non-amortized delayed execution 0.0 secs. See what I mean?

Reply:
Added `TIME_INTERFACE` to settings. Default value is `0.1`.